presto-main/src/main/java/com/facebook/presto/operator/PartitionedOutputOperator.java

```diff
@@ -433,11 +430,13 @@ public void flush(boolean force)
                     Page pagePartition = partitionPageBuilder.build();
                     partitionPageBuilder.reset();
 
+                    operatorContext.recordOutput(pagePartition.getSizeInBytes(), pagePartition.getPositionCount());
```

Comment:
@ajaygeorge Is this correct? `recordOutput` is called multiple times, once per partition with partition-specific data. Shouldn't it be called only once with a total sum across all partitions?

Reply:
@mbasmanova recordOutput calls the update() on the CounterStat which is accumulating the counts each time it's called. It can be called multiple times. 
```
public void recordOutput(long sizeInBytes, long positions)
    {
        outputDataSize.update(sizeInBytes);
        outputPositions.update(positions);
    }
```

```
public void update(long count)
    {
        oneMinute.add(count);
        fiveMinute.add(count);
        fifteenMinute.add(count);
        this.count.addAndGet(count);
    }
```

In fact in previous implementation it is called every time a page is added. In this fix it is called when flush happens but on every partition. The performance penalty calling this line on 5000 pages and forcing flush quite often is fairly small (<=3%). So comparing the fix complexity/code cleanliess/performance penalty I think this is best choice.

Reply:
Alternatively we can have two class members to track the output data sizes and rows, and update them at the same place in flush(). Then update the operatorContext stat in finish() just once. I expect the performance would be similar.